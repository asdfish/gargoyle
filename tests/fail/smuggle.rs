use gargoyle::with_guile;

fn main() {
    with_guile(|api| {
        let t = api.make(true);
        api.without_guile(|| println!("{t:?}"));
    });

    with_guile(|api| {
        let t = api.make(true);
        api.without_guile(|| {});
        println!("{t:?}");
    });
}
