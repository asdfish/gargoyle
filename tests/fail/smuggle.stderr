error[E0502]: cannot borrow `*api` as mutable because it is also borrowed as immutable
 --> tests/fail/smuggle.rs:6:9
  |
5 |         let t = api.make(true);
  |                 --- immutable borrow occurs here
6 |         api.without_guile(|| println!("{t:?}"));
  |         ^^^^-------------^^^^^^^^^^^^^^^^^^^^^^
  |         |   |
  |         |   immutable borrow later used by call
  |         mutable borrow occurs here

error[E0502]: cannot borrow `*api` as mutable because it is also borrowed as immutable
  --> tests/fail/smuggle.rs:11:9
   |
10 |         let t = api.make(true);
   |                 --- immutable borrow occurs here
11 |         api.without_guile(|| {});
   |         ^^^^^^^^^^^^^^^^^^^^^^^^ mutable borrow occurs here
12 |         println!("{t:?}");
   |                   ----- immutable borrow later used here
